### DISCUSSION TITLE
learning Container

### DISCUSSION THREAD

**Discussion by shivarm** [Hyperfoil/Horreum#719] [Q&A] [answered]
[INFO] [io.quarkus.container.image.jib.deployment.JibProcessor] Using base image with digest: sha256:9937373816f3fddf91ad55ee68576c6ab36e9e491f955393b33b4bddfead9843
[INFO] [io.quarkus.container.image.jib.deployment.JibProcessor] Container entrypoint set to [/deployments/horreum.sh]

@jesperpedersen @johnaohara Please tell me about this above thing i want to know that how project pulling images and running containers, where it implemented?

**Comment by johnaohara** [comment, marked as answer]
We are using JIB (https://cloud.google.com/blog/products/application-development/introducing-jib-build-java-docker-images-better) to build the caintiner using the `quarkus-container-image-jib` extensions:

https://github.com/Hyperfoil/Horreum/blob/master/horreum-backend/pom.xml#L117

See the quarkus guide: https://quarkus.io/guides/container-image#jib

It needs minimal configuration: https://github.com/Hyperfoil/Horreum/blob/master/horreum-backend/src/main/resources/application.properties#L94

The base image is built from this Dockerfile: https://github.com/Hyperfoil/Horreum/blob/master/horreum-backend/src/main/docker/Dockerfile.jvm.base  and pushed to quay.io

**Reply by johnaohara** [reply]
Jib allows us to build images in environments where this is no docker/podman installed

**Reply by johnaohara** [reply]
The base image is built and pushed to quay.io seperately

quay.io is Red Hats image registry, similar to docker.io

**Reply by shivarm** [reply]
Thanks john!
